### PR TITLE
Pr branch

### PR DESCRIPTION
**Pull request by rashmi-kulkarni-ct402**

### CODE REVIEW

File: Obj_To_Stl/headers/Triangle.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        int index2() const;
+        int index3() const;
+        int normalIndex() const;
+
+        void setindex1(int inIndex1);
+        void setindex2(int inIndex2);
+        void setindex3(int inIndex3);
+
+    private:
+        int mIndex1;
```
Comment (adeshlad2) [inline Obj_To_Stl/headers/Triangle.h:24]: Triangle should always store 3 Point3D vertices and 1 Point3D normal.

File: Obj_To_Stl/headers/Triangle.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include "Point3D.h"
+
+namespace Shapes3D
+{
+    class Triangle
+    {
+    public:
+        Triangle();
+        Triangle(int inIndex1, int inIndex2, int inIndex3, int inNormalIndex);
```
Comment (adeshlad2) [inline Obj_To_Stl/headers/Triangle.h]: change methods for Point3D compatibility.

File: Obj_To_Stl/headers/Triangulation.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        Triangulation();
+        ~Triangulation();
+
+        std::vector<Point3D> &uniquePoints();
+        std::vector<Triangle> &triangles();
+        std::vector<Point3D> &uniqueNormals();
+
+    private:
+        std::vector<Point3D> mUniquePoints;
+        std::vector<Triangle> mTriangles;
```
Comment (adeshlad2) [inline Obj_To_Stl/headers/Triangulation.h:23]: Triangulation should only store Triangle-vector. Unique points can be stored in different class.

File: Obj_To_Stl/headers/Triangulation.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+namespace Shapes3D
+{
+    class Triangulation
+    {
+    public:
+        Triangulation();
+        ~Triangulation();
+
+        std::vector<Point3D> &uniquePoints();
+        std::vector<Triangle> &triangles();
```
Comment (adeshlad2) [inline Obj_To_Stl/headers/Triangulation.h]: Returning reference is not recommended as mTriangles is set private. make methods to add and remove element from mTriangles.

File: Obj_To_Stl/headers/Triangulation.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace Shapes3D
+{
+    class Triangulation
+    {
+    public:
+        Triangulation();
+        ~Triangulation();
+
+        std::vector<Point3D> &uniquePoints();
```
Comment (adeshlad2) [inline Obj_To_Stl/headers/Triangulation.h]: Remove other functions. Add to different class.

File: Obj_To_Stl/src/ObjReader.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        std::istringstream vertexStream(line);
+        std::string keyword;
+        vertexStream >> keyword;
+
+        if (keyword == "v")
+        {
+            double x, y, z;
+            vertexStream >> x >> y >> z;
+            Point3D point(x, y, z);
```
Comment (adeshlad2) [inline Obj_To_Stl/src/ObjReader.cpp:39]: store this point into points vector in this function, no need for new class, as new points will always be unique.

File: Obj_To_Stl/src/ObjReader.cpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            triangulationObj.uniquePoints().push_back(point);
+        }
+        else if (keyword == "vn")
+        {
+            double x, y, z;
+            vertexStream >> x >> y >> z;
+            Point3D normal(x, y, z);
```
Comment (adeshlad2) [inline Obj_To_Stl/src/ObjReader.cpp:50]: same as point

**Review by adeshlad2 — CHANGES_REQUESTED** [review]
Comments are added to single project, required changes to be made in all projects. Will check writer later.

File: Obj_To_Stl/headers/Triangle.h (hunk 1 above)
Comment (rashmi-kulkarni-ct402) [reply Obj_To_Stl/headers/Triangle.h:24]: Triangle class has "int" variables (we are declaring indices and so it is taken as "int") as mentioned by Vinayak. I have later mapped "Point3D" and with "int" indices in Reader file.

File: Obj_To_Stl/headers/Triangle.h (hunk 2 above)
Comment (rashmi-kulkarni-ct402) [reply Obj_To_Stl/headers/Triangle.h]: Triangle class has "int" variables (we are declaring indices and so it is taken as "int") as mentioned by Vinayak. I have later mapped "Point3D" and with "int" indices in Reader file.

File: Obj_To_Stl/src/ObjReader.cpp (hunk 6 above)
Comment (rashmi-kulkarni-ct402) [reply Obj_To_Stl/src/ObjReader.cpp:39]: As we require Point3D (points) and int (indices) for mapping, these both are included in Triangulation class. The same goes with Point3D (normal) and int (indices).

File: Obj_To_Stl/src/ObjReader.cpp (hunk 7 above)
Comment (rashmi-kulkarni-ct402) [reply Obj_To_Stl/src/ObjReader.cpp:50]: As we require Point3D (points) and int (indices) for mapping, these both are included in Triangulation class. The same goes with Point3D (normal) and int (indices).

File: Obj_To_Stl/headers/Triangulation.h (hunk 4 above)
Comment (rashmi-kulkarni-ct402) [reply Obj_To_Stl/headers/Triangulation.h]: updated this by adding functions to add point, triangles into traingulation object instead of using references

**Pull request merged**